app/src/org/commcare/activities/StandardHomeActivity.java

```diff
@@ -98,9 +101,32 @@ void syncButtonPressed() {
                     AnalyticsParamValue.SYNC_MODE_SEND_FORMS,
                     AnalyticsParamValue.SYNC_FAIL_NO_CONNECTION);
             return;
+        } else {
```

Comment:
Since we are performing the granular task check here, the check above feels repeatetive. Is is possible to merge the two?